File: flutter_app/lib/models/service.dart
class ServiceModel {
  final int id;
  final String name;
  final DateTime serviceTime;
  final String type;
  final bool active;
  final bool canCheckAttendance;

  ServiceModel({
    required this.id,
    required this.name,
    required this.serviceTime,
    required this.type,
    required this.active,
    required this.canCheckAttendance,
  });

  factory ServiceModel.fromJson(Map<String, dynamic> json) {
    return ServiceModel(
      id: json['id'],
      name: json['name'],
      serviceTime: DateTime.parse(json['serviceTime']),
      type: json['type'],
      active: json['active'],
      canCheckAttendance: json['canCheckAttendance'] ?? false,
    );
  }

  String getTypeDisplayName() {
    switch (type) {
      case 'SUNDAY':
        return '주일 예배';
      case 'WEDNESDAY':
        return '수요 예배';
      case 'FRIDAY':
        return '금요 예배';
      case 'SPECIAL':
        return '특별 예배';
      default:
        return type;
    }
  }
}

File: backend/src/main/java/com/church/attendance/entity/Service.java
package com.church.attendance.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "services")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
public class Service {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, length = 100)
    private String name;  // 예: "주일 1부 예배", "수요 예배"
    
    @Column(nullable = false)
    private LocalDateTime serviceTime;  // 예배 시작 시간
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private ServiceType type = ServiceType.SUNDAY;
    
    @Column(nullable = false)
    private Boolean active = true;
    
    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @LastModifiedDate
    @Column(nullable = false)
    private LocalDateTime updatedAt;
    
    public enum ServiceType {
        SUNDAY,      // 주일 예배
        WEDNESDAY,   // 수요 예배
        FRIDAY,      // 금요 예배
        SPECIAL      // 특별 예배
    }
}


